src/modules/project-config.ts

```diff
@@ -78,10 +81,20 @@ export class ProjectConfig {
             config._variables = new Map(Object.entries(config._variables));
         }
 
+        if (!ignoreLock && ProjectConfigLock.isAvailable()) {
+            projectConfigLock = ProjectConfigLock.read();
+        }
+
         for (let packageConfig of config._packages) {
             if (packageConfig.variables) {
                 packageConfig.variables = new Map(Object.entries(packageConfig.variables));
             }
+            if (projectConfigLock) {
+                const lockedVersion = projectConfigLock.findVersion(packageConfig.repo);
+                if (lockedVersion !== undefined) {
+                    packageConfig.version = lockedVersion;
```

Comment:
Since we are overwriting the version, if we persist this packageConfig by calling PackageConfig.write(), won't we overwrite the defined range with a specific version?

Reply:
We do not support PackageConfig.write()